scripts/adaptive-sampling-integration-test.sh

```diff
@@ -2,16 +2,69 @@
 
 set -euf -o pipefail
 
-# This script is currently a placeholder.
+# Function to start Jaeger with adaptive sampling
+start_jaeger() {
+  echo "Starting Jaeger with adaptive sampling..."
+  SAMPLING_STORAGE_TYPE=memory SAMPLING_CONFIG_TYPE=adaptive go run -tags=ui ../cmd/all-in-one --log-level=debug
 
-# Commands to run integration test:
-#   SAMPLING_STORAGE_TYPE=memory SAMPLING_CONFIG_TYPE=adaptive go run -tags=ui ./cmd/all-in-one --log-level=debug
-#   go run ./cmd/tracegen -adaptive-sampling=http://localhost:14268/api/sampling -pause=10ms -duration=60m
+  # Wait for Jaeger to start (adjust sleep duration as needed)
+  until curl -s 'http://localhost:16686' > /dev/null; do
+    echo "waiting for Jaeger to start"
+    sleep 2
+  done
+  echo "Jeager started"
+}
 
-# Check how strategy is changing
-#   curl 'http://localhost:14268/api/sampling?service=tracegen' | jq .
+# Function to run trace generator with adaptive sampling
+run_trace_generator() {
+  echo "Running trace generator with adaptive sampling..."
+  go run ./cmd/tracegen -adaptive-sampling=http://localhost:14268/api/sampling -pause=10ms -duration=60m
 
-# Issues
-# - SDK does not report sampling probability in the tags the way Jaeger SDKs did
-# - Server probably does not recognize spans as having adaptive sampling without sampler info
-# - There is no way to modify target traces-per-second dynamically, must restart collector.
+  # Wait for trace generation to complete (adjust duration as needed)
+  until curl -s 'http://localhost:14268' >dev/null; do
+    echo "waiting for trace generator to start.."
+    sleep 2
+  done 
+  echo "Trace generator started."
+}
+
+# Function to check adaptive sampling strategy changes
+check_sampling_strategy() {
+  echo "Initial adaptive sampling strategy:"
+  initial_strategy=$(curl -s 'http://localhost:14268/api/sampling?service=tracegen')
+  echo "Initial adaptive sampling stratergy:"
+  echo "$initial_strategy" | jq.
+  
+  # Wait for some time to allow for adaptive sampling adjustments (adjust sleep duration as needed)
+  adjustment_duration=30
+  echo "Waiting for $adjustment_duration seconds to allow for adaptive sampling adjustments..."
+  sleep $adjustment_duration
```

Comment:
Once again, we don't just sleep for random durations. CI runs on shared VMs, it's unpredictable how much CPU quota a given runner might get (or get starved of). Sleeps are only used in small increments when checking for a specific condition in a loop.